app/views/ActionSheet.js

```diff
@@ -0,0 +1,198 @@
+		borderBottomWidth: StyleSheet.hairlineWidth,
+		borderBottomColor: COLOR_SEPARATOR,
+		flexDirection: 'row'
+	},
+	panelButtonIcon: {
+		paddingRight: 10
+	},
+	header: {
+		backgroundColor: COLOR_BACKGROUND_CONTAINER,
+		paddingTop: 15,
+		borderTopLeftRadius: 20,
+		borderTopRightRadius: 20,
+		alignItems: 'center'
+	},
+	panelHandle: {
+		width: 40,
+		height: 8,
+		borderRadius: 4,
+		backgroundColor: COLOR_SEPARATOR
+	},
+	androidButtonView: {
+		borderBottomWidth: StyleSheet.hairlineWidth,
+		borderBottomColor: COLOR_SEPARATOR
+	},
+	buttonText: {
+		...sharedStyles.textRegular,
+		fontSize: 16
+	},
+	danger: {
+		color: 'red'
+	}
+});
+
+export default class ActionSheet extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			options: [],
+			height: 100
+		};
+		this.bottomSheetRef = React.createRef();
+	}
+
+	componentDidMount() {
+		EventEmitter.addEventListener(LISTENER, this.handleDisplay);
+	}
+
+	componentWillUnmount() {
+		EventEmitter.removeListener(LISTENER);
+	}
+
+	handleDisplay = ({ options, snapPoint }) => {
+		const height = options.length * buttonPaddingSize + options.length * textSize + bottomSheetHeaderHeight;
+		this.setState({ options, height });
+		this.bottomSheetRef.current.snapTo(snapPoint);
+		vibrate();
+	}
+
+	hideActionSheet = () => {
+		this.bottomSheetRef.current.snapTo(0);
+	}
+
+
+	renderHeader = () => (
+		<View style={styles.header}>
+			<View style={styles.panelHandle} />
+		</View>
+	);
+
+	buttonIcon = (icon, isDanger) => {
+		if (!icon) {
+			return null;
+		}
+		let dangerStyle = {};
+		if (isDanger) {
+			dangerStyle = styles.danger;
+		}
+		return (
+			<CustomIcon name={icon} size={18} style={[styles.panelButtonIcon, dangerStyle]} />
+		);
+	}
+
+	buttonText = (label, isDanger) => {
+		let dangerStyle = {};
+		if (isDanger) {
+			dangerStyle = styles.danger;
+		}
+		return (
+			<Text style={[styles.buttonText, dangerStyle]}>{label}</Text>
+		);
+	}
+
+	onOptionPress = (action) => {
+		action();
+		this.hideActionSheet();
+	}
+
+	renderInner = () => {
+		const { options, height } = this.state;
+		return (
+			<View style={[styles.panel, { height }]}>
+				{options.map(option => (
+					isIOS
+						? (
+							<Touchable onPress={() => this.onOptionPress(option.handler)} key={option.label}>
+								<View style={styles.panelButton}>
+									{this.buttonIcon(option.icon, option.isDanger)}
+									{this.buttonText(option.label, option.isDanger)}
+								</View>
+							</Touchable>
+						) : (
+							<View style={styles.androidButtonView}>
+								<RectButton onPress={() => this.onOptionPress(option.handler)} key={option.label} style={styles.panelButton}>
+									{this.buttonIcon(option.icon, option.isDanger)}
+									{this.buttonText(option.label, option.isDanger)}
+								</RectButton>
+							</View>
+						)
+				))}
+			</View>
+		);
+	}
+
+	render() {
+		const { height } = this.state;
+		const snapPoints = [0, 0.25 * height, 0.5 * height, height];
```

Comment:
No need to have that many snap points.
Actions like `Resend` and `delete` message from MessageError will have only two options.
I think you can let the caller set how many snap points and where they are needed.
You can add a default as `[0, height / 2, height]`

Reply:
cool

Reply:
I cannot do it right now osdnk/react-native-reanimated-bottom-sheet#45